https/formData.go

```diff
@@ -207,6 +203,25 @@ func (fp *formParam) processSlice() (map[string][]string, error) {
 	return result, nil
 }
 
+// creates index for enums and native types in slice
+func createElemIndex(elem reflect.Value, index int) any {
```

Comment:
@haseeb-mhr This function's name/description and functionality is contradicting.

Its basically returning index for non scalar elements only, and otherwise nil. Can you name it like `getNonScalerElemIndex`, you can similarly update the description.